src/sentry/search/events/filter.py

```diff
@@ -321,6 +325,118 @@ def _team_key_transaction_filter_converter(
     )
 
 
+MAX_SEMVER_SEARCH_RELEASES = 1000
+FAKE_PACKAGE = "__sentry_fake__"
+
+
+def _flip_field_sort(field: str):
+    return field[1:] if field.startswith("-") else f"-{field}"
+
+
+def parse_semver_search(
+    search_filter: SearchFilter,
+    name: str,
+    params: Optional[Mapping[str, Union[int, str, datetime]]],
+) -> Tuple[str, str, Sequence[str]]:
```

Comment:
Nit: This function is kinda verbose, maybe not a bad idea to break it down ? like the inverse query could be in its own func and also feel like the part where `major, minor, ... etc` are extracted and then `filter_func` is created could be extracted into its own func too since `major, minor,... etc` are not used beyond https://github.com/getsentry/sentry/pull/26440/files#diff-a278bb701c83c69212137a1d12ead41fa91d73b5305dc4b528233db542811b69R380

Reply:
Yeah, I plan to break it into smaller pieces as I use it in other places. Release search will need to use parts of this without the conversion to a snuba query. I'd rather tackle this as I work on those pieces though, so that I understand the requirements better